### ISSUE TITLE
Fix error message for alpha-value-notation

### ISSUE THREAD

**Issue by icleolion** [stylelint/stylelint#4792]
> Clearly describe the bug
For whatever reason the error message for alpha-value-notation rule when set to `percentage` incorrectly expects 0.14 to be 14.000000000000002%.

> Which rule, if any, is the bug related to?
alpha-value-notation set to `percentage`

> What code is needed to reproduce the bug?
```css
.testclass {
	box-shadow: 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 1px 8px 0 rgba(0, 0, 0, 0.12) !important;
}
```

> What stylelint configuration is needed to reproduce the bug?
```json
module.exports = {
	extends: `stylelint-config-standard`,
	rules: {
		'alpha-value-notation': `percentage`,
	},
}
```

> Which version of stylelint are you using?
13.5.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?
CLI with `stylelint  **/*.vue **/*.css` and the stylelint v0.84.0 VS Code extension 

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?
standard css

> What did you expect to happen?
Error message to state "expects 0.14 to be 14%".

> What actually happened (e.g. what warnings or errors did you get)?
![image](https://user-images.githubusercontent.com/47498402/82380452-61e12f00-9a20-11ea-8df5-f57125c92d46.png)

As you can see on the same line it's managed to convert the other numbers to percentages, it's only 0.14 that's wrong throughout my whole codebase.

**Comment by m-allanson** [comment]
Hi @icleolion, thank you for using the issue template. I can [replicate this bug on the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYKIAclgJYBcBPeFABgDoBGAFhQF8QAaEAZ0IDt8kAPBcFlk3AR2AM1wBzPsAA67FChkgATgFcANjBZKSs+QoVKk6zAAskAWgBux1TAvsIXfLhE7FITDGVgYnJBIwSnIKdHIMdEA).

It looks like [the rule's `asNumber` function](https://github.com/stylelint/stylelint/blob/a859f1ae08b2247f6116a08e1848cf44234bfc30/lib/rules/alpha-value-notation/index.js#L124) needs some guards on its calculation.

Something like this should do it:

```diff
// in lib/rules/alpha-value-notation/index.js

function asNumber(value) {
	const { number } = valueParser.unit(value);

-	return number / 100;
+	return Number((number / 100).toFixed(2));
}
```

(based on [this SO answer](https://stackoverflow.com/questions/12227594/which-is-better-numberx-or-parsefloatx/13676265#13676265))

The [test cases](https://github.com/stylelint/stylelint/blob/master/lib/rules/alpha-value-notation/__tests__/index.js) will also need updating. It could be worth investigating whether the `asPercentage` function should have similar guards added.

A PR would be muchly appreciated if you'd like to contribute this fix. No worries if not, I'll mark this issue as ready to implement.

**Comment by icleolion** [comment]
Sorry, my workplace has me stuck in TFVC land.  :unamused:

Learning to work with git and github represents a barrier to entry that I can't overcome with the free-time that I have (between family and work that would be none).

I'm afraid reporting bugs is as much as I'll be able to "contribute".

**Comment by m-allanson** [comment]
That is understandable and not a problem! Thank you for the clear bug report - clear reports make it much easier to track down the source of a bug.

**Comment by malsf21** [comment]
I can probably implement this on the weekend - I'll add the type coercion + `.toFixed(2)` output. I can also look into `asPercentage` later if I have time!
